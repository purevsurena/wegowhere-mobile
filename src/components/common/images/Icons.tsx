import React from 'react';
import { SvgXml } from 'react-native-svg';
import { scale } from '@/utils/Metrics';
import { IconTypeProps } from '@/types/icons';
import IConstant from '@/constants/icons';

function Icons({ name, width, height, styles }: IconTypeProps) {
  const src = name && IConstant[name];
  return (
    <SvgXml
      testID={`icon-${name}`}
      width={scale(width)}
      height={scale(height)}
      xml={`${src}`}
      style={{ ...styles }}
    />
  );
}

export default Icons;
