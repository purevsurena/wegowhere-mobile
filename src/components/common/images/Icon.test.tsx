// Icons.test.js

import React from 'react';
import { render } from '@testing-library/react-native';
import renderer from 'react-test-renderer';
import Icons from '@/components/common/images/Icons';

describe('Icons Component', () => {
  it('renders correctly', () => {
    const mockIcon = 'addIcon';
    const { getByTestId } = render(
      <Icons name={mockIcon} width={24} height={24} styles={{}} />,
    );

    const renderedIcon = getByTestId(`icon-${mockIcon}`);

    expect(renderedIcon).toBeTruthy();

    const tree = renderer
      .create(<Icons name={mockIcon} width={24} height={24} styles={{}} />)
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});
