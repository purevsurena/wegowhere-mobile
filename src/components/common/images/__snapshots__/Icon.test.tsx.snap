// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Icons Component renders correctly 1`] = `
<RNSVGSvgView
  align="xMidYMid"
  bbHeight="48"
  bbWidth="48"
  fill="none"
  focusable={false}
  height={48}
  meetOrSlice={0}
  minX={0}
  minY={0}
  style={
    [
      {
        "backgroundColor": "transparent",
        "borderWidth": 0,
      },
      {},
      {
        "flex": 0,
        "height": 48,
        "width": 48,
      },
    ]
  }
  testID="icon-addIcon"
  vbHeight={24}
  vbWidth={24}
  width={48}
  xml="<svg width="24" height="24" viewBox="0 0 24 24" fill="none" xmlns="http://www.w3.org/2000/svg">
<path d="M22.5415 10.8916H13.1083V1.45835C13.1083 0.846217 12.6121 0.35 11.9999 0.35C11.3878 0.35 10.8916 0.846217 10.8916 1.45835V10.8917H1.45836C1.45836 10.8917 1.45835 10.8917 1.45835 10.8917C0.846207 10.8916 0.35 11.3878 0.35 12C0.35 12.6121 0.846215 13.1083 1.45835 13.1083H10.8917V22.5416C10.8917 23.1538 11.3879 23.65 12 23.65C12.6121 23.65 13.1083 23.1538 13.1083 22.5416V13.1083H22.5417C23.1538 13.1083 23.65 12.6121 23.65 12C23.65 11.3878 23.1537 10.8916 22.5415 10.8916Z" fill="black" stroke="black" stroke-width="0.3"/>
</svg>
"
  xmlns="http://www.w3.org/2000/svg"
>
  <RNSVGGroup
    fill={null}
    propList={
      [
        "fill",
      ]
    }
  >
    <RNSVGPath
      d="M22.5415 10.8916H13.1083V1.45835C13.1083 0.846217 12.6121 0.35 11.9999 0.35C11.3878 0.35 10.8916 0.846217 10.8916 1.45835V10.8917H1.45836C1.45836 10.8917 1.45835 10.8917 1.45835 10.8917C0.846207 10.8916 0.35 11.3878 0.35 12C0.35 12.6121 0.846215 13.1083 1.45835 13.1083H10.8917V22.5416C10.8917 23.1538 11.3879 23.65 12 23.65C12.6121 23.65 13.1083 23.1538 13.1083 22.5416V13.1083H22.5417C23.1538 13.1083 23.65 12.6121 23.65 12C23.65 11.3878 23.1537 10.8916 22.5415 10.8916Z"
      fill={
        {
          "payload": 4278190080,
          "type": 0,
        }
      }
      propList={
        [
          "fill",
          "stroke",
          "strokeWidth",
        ]
      }
      stroke={
        {
          "payload": 4278190080,
          "type": 0,
        }
      }
      strokeWidth="0.3"
    />
  </RNSVGGroup>
</RNSVGSvgView>
`;
